src/components/CompanySettingsRoom.js

```diff
@@ -1,73 +1,55 @@
-import React from 'react';
-
-const getRoomNameFromInput = parentNode => {
-  let roomName = '';
-  parentNode.childNodes.forEach(childNode => {
-    if (childNode.name === 'roomName') {
-      roomName = childNode.value;
-    }
-  });
-  return roomName;
-};
-
-const handleUpdateClick = (updateRoomForHotel, id, hotelId) => {
-  return event => {
-    event.preventDefault();
-    const roomName = getRoomNameFromInput(event.target.parentNode);
-    if (window.confirm('Are you sure you want to edit this room?')) {
-      if (roomName) {
-        updateRoomForHotel(hotelId, id, roomName);
-      }
-    }
-  };
-};
-
-const handleDeleteClick = (deleteRoomForHotel, id, hotelId) => {
-  return event => {
-    if (window.confirm('Are you sure you want to delete this room?')) {
-      event.preventDefault();
-      deleteRoomForHotel(id, hotelId);
-    }
-  };
-};
+import React, { useState } from 'react';
+import Confirm from './reusable/ConfirmModal';
 
 const CompanySettingsRoom = ({
   room,
-  rooms,
   index,
   name,
   currentUser,
-  handleRoomInputChange,
   deleteRoomForHotel,
   updateRoomForHotel,
 }) => {
+  const [isDeleteModalOpen, setIsDeleteModalOpen] = useState(false);
+  const [isUpdateModalOpen, setIsUpdateModalOpen] = useState(false);
+  const [inputValue, setInputValue] = useState(name);
+  const handleDeleteClick = () => {
+    setIsDeleteModalOpen(true);
+  };
+  const handleUpdateClick = () => {
+    setIsUpdateModalOpen(true);
+    // updateRoomForHotel(hotelId, id, roomName);
+  };
+
   return !room || !room[index] ? (
```

Comment:
guessing we can remove room[index] from here now and the arguments list for CompanySettingsRoom? and also from the map on the parent component?